spot_driver_cpp/include/spot_driver_cpp/conversions/common_conversions.hpp

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2023 Boston Dynamics AI Institute LLC. All rights reserved.
+
+#pragma once
+
+#include <google/protobuf/timestamp.pb.h>
+#include <google/protobuf/wrappers.pb.h>
+
+#include <bosdyn_msgs/msg/arm_joint_position.hpp>
+#include <bosdyn_msgs/msg/common_error.hpp>
+#include <bosdyn_msgs/msg/quaternion.hpp>
+#include <bosdyn_msgs/msg/request_header.hpp>
+#include <bosdyn_msgs/msg/response_header.hpp>
+
+#include <bosdyn/api/arm_command.pb.h>
+#include <bosdyn/api/geometry.pb.h>
+#include <bosdyn/api/header.pb.h>
+
+#include <geometry_msgs/msg/point.hpp>
+#include <geometry_msgs/msg/pose.hpp>
+#include <geometry_msgs/msg/quaternion.hpp>
+#include <geometry_msgs/msg/twist.hpp>
+#include <geometry_msgs/msg/vector3.hpp>
+
+namespace spot_ros2::common_conversions {
+
+///////////////////////////////////////////////////////////////////////////////
+// ROS to Protobuf.
+
+void convertBuiltinInterfacesTimeToProto(const builtin_interfaces::msg::Time& ros_msg,
+                                         google::protobuf::Timestamp& proto);
```

Comment:
@gremigi-bdai and I talked about this offline. He pointed our a few challenges with this change. For example, some proto messages correspond to multiple ROS messages, so the parameter type isn't sufficient to disambiguate them. In these cases the function would need to be named differently anyway (or some other architectural choice would need to be made here).

For the time being I'm OK with keeping the current function signatures that use output parameters, partly because aside from whatever the style guide says this isn't functionally _that_ big of a deal and it's more important to keep this work moving along.

Let's keep ourselves open to revising that decision later on, though. I expect this will come up again in the context of autogenerating C++-based protobuf-to-ROS conversions.

Reply:
As per our discussion, I renamed all methods into "convertToProto" and "convertToRos".